[He Shi] ip

lancehaha opened the pull request:
# Chatgpt

>[User] todo assignment
>[Chatgpt] got it!
>[User] list
>[Chatgpt] 1, [] [] assignment

Chatgpt is a chatbot for.memorizing the finished and unfinished tasks. Still working on to make it better 😄 

Features:

- Text-based
- Easy to learn
- FAST ~~SUPER FAST~~ to use!

All you need to do is:

try it from [here](https://github.com/lancehaha/ip)
1. Double-click it.
2. Add your tasks
3. Let it manage your tasks for you
It can:

- [x] Manage todos, deadlines and events
- [ ] interact via a GUI (coming soon!)

```
public static void main(String[] args) {

    Tars tars = new Tars();
    tars.run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lancehaha committed:
level-0

lancehaha committed:
level-0

lancehaha committed:
level-0

lancehaha committed:
level-4 and i forget to change name for the last 3 commits

lancehaha committed:
level-5

lancehaha committed:
level-6

lancehaha committed:
automate testing

lancehaha committed:
A-TextUiTesting

lancehaha committed:
automate testing

lancehaha committed:
Merge remote-tracking branch 'origin/master'

lancehaha committed:
A-Enums

lancehaha committed:
Level-0

lancehaha committed:
Level-1

lancehaha committed:
Level-2

lancehaha committed:
Level-2

lancehaha committed:
Level-4

lancehaha committed:
Level-5

lancehaha committed:
Level-6

lancehaha committed:
Level-7 finished add file access and directory creation access

automatic update added

lancehaha committed:
just random commit to try to figure out merge process

lancehaha committed:
just random commit to try to figure out merge process

lancehaha committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  just random commit to try to figure out merge process

lancehaha committed:
just random commit to try to figure out merge process

lancehaha committed:
I forget to create branch-level-8

lancehaha committed:
I forget to create branch-level-8

lancehaha committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  I forget to create branch-level-8

lancehaha committed:
created more objects to integrate the functions, and put them into packages

lancehaha committed:
created more objects to integrate the functions, and put them into packages

lancehaha committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

lancehaha committed:
A-JUnit creating junit test for todo and deadline

lancehaha committed:
A-Jar finished

lancehaha committed:
A-Jar finished

lancehaha committed:
Level-9

lancehaha committed:
Level-9

lancehaha committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

lancehaha committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Level-9

Siddardar commented on src/main/java/Chatgpt.java:
```diff
@@ -0,0 +1,110 @@
+package Tasks;
```
I believe names representing packages should be in [lowercase](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html).

Siddardar commented on src/main/java/Chatgpt.java:
```diff
@@ -0,0 +1,110 @@
+     * @param args Command line arguments passed to the application.
+     * @throws EmptyDescriptionException if the task description is missing.
+     */
+    public static void main(String[] args) throws EmptyDescriptionException {
```
I like the detailed custom exception classes 👍

Siddardar commented on src/main/java/Tools/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+        }
+        String description = input.substring(5).trim();
+        tasks.add(new Todo(description));
+        System.out.println("Got it. I've added this task:");
```
Can consider using a text block along with `String.format` instead to make the message a bit clearer

Siddardar commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,28 @@
+ * toFileFormat method. It verifies that the method returns the correct string format for storing
+ * deadline tasks in a file.
+ */
+class DeadlineTest {
```
Can consider testing via user input rather then just creating a `Deadline` object. Maybe you could test creating a `Deadline` object by executing a `AddCommand`.

Siddardar commented on src/main/java/Tools/Parser.java:
```diff
@@ -0,0 +1,52 @@
+        this.ui = ui;
+    }
+
+    public void parse(String input) {
```
Could consider a switch statement here by extracting `input.startsWith`.

Siddardar reviewed (commented):
Overall pretty good 👍 

Nice one 💯

lancehaha committed:
Adding checkstyle into the system

but I haven't modified my original code to make it perfectly fitting the style yet

lancehaha committed:
new window of UI successfully designed but for saving and adding seems still having problems

struggling on how to solve that

lancehaha committed:
the level-10 finished!!!

I have to store the changes in a file after every command in order to solve my problem weird!!!!

lancehaha committed:
merge commit

add the images I forgot to commit and also for merge the branches

lancehaha committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  the level-10 finished!!!
  new window of UI successfully designed but for saving and adding seems still having problems

lancehaha committed:
merge commits in master branch

this commits is add so that when I add a future commit it wouldn't be a fastforward merge

no significant change

lancehaha committed:
assert commit to check if there is any obvious error

assert function are added to check if the adding and deleting functions are work finely

lancehaha committed:
codequality commit that optimized the code

some of the functions in newStorage looks too ugly too long and too many things were happening at the same time not really readable, so I optimized it

lancehaha committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  assert commit to check if there is any obvious error

lancehaha committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  codequality commit that optimized the code

lancehaha committed:
pull request merge failure

this meaningless commit didn't change anything but to let me experiment with merge pr function

lancehaha committed:
pull request merge failure

this meaningless commit didn't change anything but to let me experiment with merge pr function, but also changed a bit of format about switch and case

lancehaha committed:
Merge pull request #1 from lancehaha/branch-A-CodeQuality

branch-A-CodeQuality

lancehaha committed:
Merge pull request #2 from lancehaha/branch-A-Assertions

branch-A-Assertion

lancehaha committed:
BCD-Extension I chose B-update and C-priority. The code in TaskList and Paser and all 4 kinds of the tasks were changed to include the priority attribute and the update function

I extended more function of my code to make it more powerful and also to complete the BCD extension task

lancehaha committed:
updated a lot of the function

the lost BCD-extentions are added right here

lancehaha committed:
updated a lot of the function

the lost BCD-extentions are added right here

lancehaha committed:
Revert "updated a lot of the function"

This reverts commit d1d18eac050e4699593cea8b2f8ae79d1b5c2546.

lancehaha committed:
no message

lancehaha committed:
Merge branch 'random-testing-branch'

* random-testing-branch:
  Revert "updated a lot of the function"

lancehaha committed:
A-UserGuide

finished the user guide readme

the ip is finished for better user experience I designed the doc readme as a hand book for user

hope this doc can help with the user when they are confused.

lancehaha committed:
docs: add user guide for project setup and usage

Include detailed instructions covering:
-features of my project
-command for using feature

This guide aims to assist new users in setting up and effectively using the application.

lancehaha committed:
Merge branch 'master' of https://github.com/lancehaha/ip

* 'master' of https://github.com/lancehaha/ip:
  A-UserGuide

# Conflicts:
#	docs/README.md

lancehaha committed:
Fix priority display bug in NewTaskList.java

Resolve an issue where the priority was not being displayed correctly in the NewTaskList component. This fix ensures that priority levels are shown as intended, improving UI accuracy and user interaction.

lancehaha committed:
docs: add UI screenshot to the documentation folder

A screenshot illustrating the user interface has been added to the 'doc' folder. This image helps provide a visual representation of the application interface, making it easier for users to understand the layout and functionality as documented on GitHub.

lancehaha committed:
docs: add Javadoc comments to NewParser methods

Add comprehensive Javadoc descriptions to methods in the NewParser class. These comments explain the functionality, parameters, and expected outcomes of each method, improving code readability and maintainability. The update facilitates easier navigation and understanding of the codebase for future development and reviews.

lancehaha committed:
add method comments to Task objects

Add method comment to the Task object for the user to have a better understanding and for myself to better use the method

lancehaha committed:
docs:change UI.png name to Ui.png

change the Ui web name for user to browse the ui page

lancehaha committed:
docs: add Javadoc comments to NewStorage class methods

Added detailed Javadoc descriptions to all methods in the NewStorage class. These comments explain the functionality, parameters, and expected behavior of each method, improving code readability and maintainability.

lancehaha committed:
docs: add Javadoc comments to NewTaskList class methods

Add detailed Javadoc descriptions for all methods in the NewTaskList class. These comments elucidate the purpose, parameters, exceptions, and return values of each method, enhancing code readability and facilitating easier maintenance and usage by developers.

lancehaha committed:
docs: add Javadoc comments to Parser methods

Add comprehensive Javadoc descriptions to methods in the Parser class. These comments explain the functionality, parameters, and expected outcomes of each method, improving code readability and maintainability. The update facilitates easier navigation and understanding of the codebase for future development and reviews.

lancehaha committed:
add Javadoc comments to Storage class methods

Added detailed Javadoc descriptions to all methods in the Storage class. These comments explain the functionality, parameters, and expected behavior of each method, improving code readability and maintainability.

lancehaha committed:
add Javadoc comments to TaskList class methods

Added detailed Javadoc descriptions to all methods in the TaskList class. These comments explain the functionality, parameters, and expected behavior of each method, improving code readability and maintainability.

change the Storage comments a bit

the storage comments are changed to indicate it is the abolished version for chatbot's version without ui

lancehaha committed:
docs: add Javadoc comments to Ui class methods

Enhance the Ui class by adding detailed Javadoc descriptions for each method. These comments clarify the purpose, usage, and return values of the methods, thereby improving code readability and aiding in maintenance. Also corrected a minor typo in the welcome message for better user interaction.

Reactivate the chatgpt without ui version

I copied and pasted the code of the old version chatgpt back to the program

lancehaha committed:
change build gradle file

now build.gradle bug is fixed, it creates jar file for Main.java instead of ChatgptMoreOOP.java

lancehaha committed:
change doc readme.md file

now I update the readme.md file to include prerequiste and activation step of how to test the app

lancehaha committed:
change doc build.gradle file

now I have solved a problem of jar file running on windows

lancehaha committed:
Update README.md
